aws-lc-rs/src/unstable/kdf.rs

```diff
@@ -0,0 +1,459 @@
+                SskdfHmacAlgorithmId::Sha512,
+            ] {
+                let alg = get_sskdf_hmac_algorithm(id).expect("supported");
+                assert_eq!(id, alg.id());
+                assert_eq!(format!("{id:?}"), format!("{alg:?}"));
+                assert_eq!(format!("{id:?}"), format!("{:?}", alg.id()));
+                let mut output = vec![0u8; 32];
+                sskdf_hmac(alg, &[1u8; 32], &[2u8; 32], &[3u8; 32], &mut output).expect("success");
+            }
+        }
+
+        #[test]
+        fn kbkdf_ctr_hmac_test() {
+            for id in [
+                KbkdfCtrHmacAlgorithmId::Sha224,
+                KbkdfCtrHmacAlgorithmId::Sha256,
+                KbkdfCtrHmacAlgorithmId::Sha384,
+                KbkdfCtrHmacAlgorithmId::Sha512,
+            ] {
+                let alg = get_kbkdf_ctr_hmac_algorithm(id).expect("supported");
+                assert_eq!(id, alg.id());
+                assert_eq!(format!("{id:?}"), format!("{alg:?}"));
+                assert_eq!(format!("{id:?}"), format!("{:?}", alg.id()));
+                let mut output = vec![0u8; 32];
+                kbkdf_ctr_hmac(alg, &[1u8; 32], &[2u8; 32], &mut output).expect("success");
+            }
+        }
+
+        #[test]
+        fn algorithm_equality() {
+            let alg1 = get_kbkdf_ctr_hmac_algorithm(KbkdfCtrHmacAlgorithmId::Sha256).unwrap();
+            let alg2 = get_kbkdf_ctr_hmac_algorithm(KbkdfCtrHmacAlgorithmId::Sha256).unwrap();
+            assert_eq!(alg1, alg2);
+            let alg2 = get_kbkdf_ctr_hmac_algorithm(KbkdfCtrHmacAlgorithmId::Sha512).unwrap();
+            assert_ne!(alg1, alg2);
+
+            let alg1 = get_sskdf_digest_algorithm(SskdfDigestAlgorithmId::Sha256).unwrap();
+            let alg2 = get_sskdf_digest_algorithm(SskdfDigestAlgorithmId::Sha256).unwrap();
+            assert_eq!(alg1, alg2);
+            let alg2 = get_sskdf_digest_algorithm(SskdfDigestAlgorithmId::Sha512).unwrap();
+            assert_ne!(alg1, alg2);
+
+            let alg1 = get_sskdf_hmac_algorithm(SskdfHmacAlgorithmId::Sha256).unwrap();
+            let alg2 = get_sskdf_hmac_algorithm(SskdfHmacAlgorithmId::Sha256).unwrap();
+            assert_eq!(alg1, alg2);
+            let alg2 = get_sskdf_hmac_algorithm(SskdfHmacAlgorithmId::Sha512).unwrap();
+            assert_ne!(alg1, alg2);
+        }
+    }
+
+    #[cfg(feature = "fips")]
+    mod fips {
+        use crate::unstable::kdf::{
+            get_kbkdf_ctr_hmac_algorithm, get_sskdf_digest_algorithm, get_sskdf_hmac_algorithm,
+            KbkdfCtrHmacAlgorithmId, SskdfDigestAlgorithmId, SskdfHmacAlgorithmId,
+        };
+
+        macro_rules! assert_get_unsupported_algorithm {
+            ($name:ident, $getter:path, $alg:expr) => {
+                #[test]
+                fn $name() {
+                    assert_eq!(None, $getter($alg));
+                }
+            };
+        }
+
+        assert_get_unsupported_algorithm!(
```

Comment:
Very minor nit, is there a better way to iterate over these with something like a macro?

Reply:
Yeah, it is possible to have a [variadic macro](https://doc.rust-lang.org/rust-by-example/macros/variadics.html), but given the subtle differences between each of the macro invocations below, a variadic mecro for this would likely be much less pleasant.